<?xml version="1.0" encoding="UTF-8"?>
<launch>
    <!-- gazebo launch with pre-built world including the spawning of the robot and its URDF information -->
    <include file ="$(find autonomous_robots)/launch/my_world.launch"/>

    <node pkg="teleop_twist_keyboard" type="teleop_twist_keyboard.py" name="teleop_twist_keyboard">
    </node>
    
    <!-- using the map we saved earlier -->
    <arg name="map_file" value="$(find autonomous_robots)/map/my_map_2.yaml"/>

    <!-- Map server package is used to handle the map we saved -->
    <node pkg="map_server" type="map_server" name="map_server" args="$(arg map_file)" />

    <!-- Adaptive Monte Carlo node running -->
    <include file="$(find turtlebot3_navigation)/launch/amcl.launch"/>

    <!-- publisher transform (TFs) of all joints and their inbetween relations -->
    <node pkg= "robot_state_publisher" type="robot_state_publisher" name="robot_state_publisher">
        <param name="publish_frequency" type="double" value="50.0"/>
    </node>

    <!-- Setting up the move_base node and costmap parameters -->
    <node pkg="move_base" type="move_base" name="move_base" respawn="false" output="screen">

        <param name="base_local_planner" value="dwa_local_planner/DWAPlannerROS"/>

        <rosparam file="$(find turtlebot3_navigation)/param/dwa_local_planner_params_waffle_pi.yaml" command="load"/>
        <rosparam file="$(find autonomous_robots)/params/global_planner.yaml" command="load"/>
        <rosparam file="$(find turtlebot3_navigation)/param/move_base_params.yaml" command="load"/>

        <!-- observation sources located in costmap_common.yaml -->
        <rosparam file="$(find turtlebot3_navigation)/param/costmap_common_params_waffle_pi.yaml" command="load" ns="global_costmap" />
        <rosparam file="$(find turtlebot3_navigation)/param/costmap_common_params_waffle_pi.yaml" command="load" ns="local_costmap" />

        <!-- local costmap, needs size -->
        <rosparam file="$(find turtlebot3_navigation)/param/local_costmap_params.yaml" command="load" />

        <!-- static global costmap, static map provides size -->
        <rosparam file="$(find turtlebot3_navigation)/param/global_costmap_params.yaml" command="load"/>

        <remap from="cmd_vel" to="/cmd_vel"/>
        <remap from="odom" to="/odom"/>

    </node>

</launch>

 